mscore/editstaff.cpp

```diff
@@ -347,16 +343,32 @@ void EditStaff::apply()
 
       QString newPartName = partName->text().simplified();
 
-      bool instrumentFieldChanged = !(instrument == *part->instrument());
-      if (instrumentFieldChanged)
+      bool instrumentFieldChanged = !(instrument == *part->instrument(_tickStart));
+      if (instrumentFieldChanged && _tickStart == Fraction(-1, 1))
             clefType = instrument.clefType(orgStaff->rstaff());
 
       if (instrumentFieldChanged || part->partName() != newPartName) {
             // instrument has changed
             Interval v1 = instrument.transpose();
-            Interval v2 = part->instrument()->transpose();
+            Interval v2 = part->instrument(_tickStart)->transpose();
 
-            score->undo(new ChangePart(part, new Instrument(instrument), newPartName));
+            if (_tickStart == Fraction(-1, 1)) {
```

Comment:
Which user scenario does run this branch of code?

Reply:
This should be hit if there are no instrument changes at all.  We have made the distinction between tick -1 and tick 0 for this reason elsewhere as well.